### DISCUSSION TITLE
How to update peer dependencies in lerna@7?

### DISCUSSION THREAD

**Discussion by nicojs** [lerna/lerna#3749] [Q&A]
It seems that `lerna version` also runs `npm install --package-lock-only --ignore-scripts` during its process. This results in an error since lerna@7, because npm workspaces don't support a peer dep mismatch. I believe the old `lerna bootstrap` command did allow it.

<details>
<summary>See detailed logs</summary>

```
GH_TOKEN=$(gh auth token) npx lerna version
lerna notice cli v7.0.2
lerna info current version 7.0.20
lerna WARN force-publish all packages
lerna info Assuming all packages changed
lerna info getChangelogConfig Successfully resolved preset "conventional-changelog-angular"

Changes:
 - @stryker-mutator/api: 7.0.2 => 7.1.0
 - @stryker-mutator/core: 7.0.2 => 7.1.0
 - @stryker-mutator/cucumber-runner: 7.0.2 => 7.1.0
 - grunt-stryker: 7.0.2 => 7.1.0
 - @stryker-mutator/instrumenter: 7.0.2 => 7.1.0
 - @stryker-mutator/jasmine-runner: 7.0.2 => 7.1.0
 - @stryker-mutator/jest-runner: 7.0.2 => 7.1.0
 - @stryker-mutator/karma-runner: 7.0.2 => 7.1.0
 - @stryker-mutator/mocha-runner: 7.0.2 => 7.1.0
 - @stryker-mutator/tap-runner: 7.0.2 => 7.1.0
 - @stryker-mutator/test-helpers: 7.0.2 => 7.1.0 (private)
 - @stryker-mutator/typescript-checker: 7.0.2 => 7.1.0
 - @stryker-mutator/util: 7.0.2 => 7.1.0
 - @stryker-mutator/vitest-runner: 7.0.2 => 7.1.0

? Are you sure you want to create these versions? Yes
lerna info execute Creating releases...
lerna ERR! Error: Command failed with exit code 1: npm install --package-lock-only --ignore-scripts
lerna ERR! npm ERR! code ERESOLVE
lerna ERR! npm ERR! ERESOLVE could not resolve
lerna ERR! npm ERR!
lerna ERR! npm ERR! While resolving: @stryker-mutator/cucumber-runner@7.1.0
lerna ERR! npm ERR! Found: @stryker-mutator/core@7.0.2
lerna ERR! npm ERR! node_modules/@stryker-mutator/core
lerna ERR! npm ERR!   workspace packages/core from the root project
```

</details>

Our versioning workflow pre lerna@7 looked like this:

1. Update versions in package.json's & generate changelogs
2. Update peer dependencies
3. Commit & tag release

For step 2, we created a custom script: `"version": "node tasks/bump-peer-deps.js"` in root package.json. In that script we update the peer deps when needed.

How can I recreate such a thing in lerna@7?

**Comment by fahslaj** [comment]
Hi @nicojs , there is no lifecycle hook for Lerna version that allows you to inject a script before that installation step.

As a workaround, you can use `--legacy-peer-deps` in lerna.json's configuration for the version command, allowing it to continue past this install step (ignoring peer dependency failures) and get to your node script. If doing this, you may need to call `npm install --package-lock-only --ignore-scripts` again in your script to update the package-lock file after your manual peer dependency updates.

The lerna.json configuration update would look like this:
```json
{
...
  "command": {
    ...
    "version": {
      ...,
      "npmClientArgs": ["--legacy-peer-deps"]
    }
  }
}
```

**Reply by nicojs** [reply]
Thanks! I'll give it a try. It seems like a workaround, though, could we change `lerna` so it would run `npm install --package-lock-only --ignore-scripts` after te `version` script? Or add a new lifecycle hook (`"preprepareinstall"` or something)?

**Reply by fahslaj** [reply]
It makes sense to do the install before the version script, as the intention is that all changes that Lerna will make to the filesystem are done by the time the version lifecycle script is run. Though you've got a good point in that a new lifecycle hook for this case might be warranted.
